crates/matrix-sdk/src/encryption/mod.rs

```diff
@@ -2017,4 +2027,39 @@ mod tests {
         let after_taking_lock_second_time = client.olm_machine().await.as_ref().unwrap().clone();
         assert!(after_taking_lock_first_time.same_as(&after_taking_lock_second_time));
     }
+
+    #[async_test]
+    async fn test_update_verification_state_is_updated_before_any_requests_happen() {
+        // Given a client and a server
+        let client = no_retry_test_client(None).await;
+        let server = MockServer::start().await;
+
+        // When we subscribe to its verification state
+        let mut verification_state = client.encryption().verification_state();
+
+        // We can get its initial value, and it's Unknown
+        assert_next_matches_with_timeout!(verification_state, VerificationState::Unknown);
+
+        // We set up a mocked request to check this endpoint is not called before
+        // reading the new state
+        let keys_requested = Arc::new(AtomicU8::new(0));
+        let inner_counter = keys_requested.clone();
+        Mock::given(method("GET"))
+            .and(path_regex(
```

Comment:
You can add `set_delay()` for this mock to make it a bit more certain that we'll not respond in time: https://docs.rs/wiremock/latest/wiremock/struct.ResponseTemplate.html#method.set_delay

Reply:
But that just makes the actual response be slower, right? The code changing the counter would still be run before the delay.

Reply:
Ah duh, I just understood what you said, nevermind with that setting then.